File: 02_C++/Tasks/T2.cc
/*******************************************************
* Author: Abdullah Abdelhakeem
* Date: September 23, 2023
* Description: Maximum number between three values
*
* Type: Task
* Language: C++
* Dependencies: None
* IDE: VS Code
* Compiler: G++
* OS: Windows 10
*
* Inputs: 3 values
* Outputs: maximum
*
* Algorithm:
*   1.
*
* Note:
*******************************************************/


#include<iostream>
#include<algorithm>
int main(){
    int x,y,z;
    std::cout << "Plaese enter 3 Values : "; // ex 5  6  7
    std::cin >> x >> y >>z;
    std::cout << std::max({x,y,z});
    return 0;
}

// #include<iostream>
// #include<algorithm>
// #include<vector>
// int main(){
//     std::vector<int>v {1,2,3};
//     std::cout << *(std::max_element(v.begin(), v.end()));

//     return 0;
// }

File: 01_Python/Tasks/T2.py
# /*******************************************************
# * Author: Abdullah Abdelhakeem
# * Date: October 7, 2023
# * Description: Write a python program to test whether a passed letter is a vowel or not
# *
# * Type: Task
# * Language: Python
# * Dependencies: None
# * IDE: VS Code
# * Compiler: interpreter
# * OS: Windows 10
# *
# * Inputs:
# * Outputs:
# *
# * Algorithm:
# *   1.
# *
# * Note:
# *******************************************************/

def isVowel(letter):
    letter = letter.lower()
    if letter in ['a','e','i','o','u']:
        return True
    else:
        return False

def main():
    letter_to_test = input("Enter a letter: ")
    if len(letter_to_test)==1:
        if isVowel(letter_to_test):
            print(f"the letter '{letter_to_test}' is a vowel")
        else:
            print(f"the letter '{letter_to_test}' is not a vowel")
    else:
        print("please enter a single character")

main()

